File: others/2023_ee_night/editor-server/src/resolvers/subscriptions/controlMap.ts
import { ObjectType, Field, Int } from "type-graphql";
import { ControlDataScalar } from "../types/controlData";

@ObjectType()
export class ControlMapPayload {
  @Field((type) => ControlDataScalar)
  frame: any;

  @Field((type) => Int)
  editBy: number;
}

File: editor-server/src/graphql/subscriptions/control_map.rs
//! Color subscription methods.

use crate::graphql::subscriptor::Subscriptor;
use crate::graphql::types::control_data::ControlFramesSubDatScalar;

use async_graphql::{SimpleObject, Subscription};
use futures_core::stream::Stream;

#[derive(SimpleObject, Clone, Default)]
pub struct ControlMapPayload {
    pub frame: ControlFramesSubDatScalar,
    pub edit_by: i32,
}

#[derive(Default)]
pub struct ControlMapSubscription;

#[Subscription]
impl ControlMapSubscription {
    async fn control_map_subscription(&self) -> impl Stream<Item = ControlMapPayload> {
        Subscriptor::<ControlMapPayload>::subscribe()
    }
}
